sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -137,6 +138,9 @@
 
   @Override
   public Span.Builder setAttribute(String key, String value) {
+    if (StringUtils.isNullOrBlank(value)) {
+      return this;
+    }
     return setAttribute(key, AttributeValue.stringAttributeValue(value));
```

Comment:
What is the behavior for AttributeValue.stringAttributeValue(value) in case a null string is passed?

Reply:
that remains the same. If you're trying to create an AttributeValue with a null, you'll get the usual NPE.

Reply:
k

Reply:
I think this is unexpected result and inconsistent, maybe we should try to see how we can offer the same behavior.

Reply:
that's a fair comment. I think we can do this by allowing null/empty in the `AttributeValue`, and then ignoring when we try to add. I'll change this PR to do it that way.

Reply:
ok, changed